pinot-tools/src/main/java/org/apache/pinot/tools/data/generator/TemplateType.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.tools.data.generator;
+
+public enum TemplateType {
```

Comment:
Apologies for my ignorance, is `Template` a generally accepted term here? If not, it is not intuitive enough, and may be PATTERN/DISTRIBUTION is a better word?

Also, would be good to add javadoc explaining each type.

Reply:
Renamed to "Pattern". Definitely more appropriate since those aren't physical "templates". 

Each type already explained in the generator module. Added javadoc referencing that information.